tests/integration/test_diff.py

```diff
@@ -0,0 +1,388 @@
+"""
+These tests compare the output of the diff command with the expected output.
+"""
+
+import sys
+
+import pytest
+import vorta.borg
+import vorta.utils
+import vorta.views.archive_tab
+from pkg_resources import parse_version
+from vorta.borg.diff import BorgDiffJob
+from vorta.views.diff_result import (
+    ChangeType,
+    DiffTree,
+    FileType,
+    ParseThread,
+)
+
+
+@pytest.mark.parametrize(
+    'archive_name_1, archive_name_2, expected',
+    [
+        (
+            'test-archive1',
+            'test-archive2',
+            [
+                {
+                    'subpath': 'dir',
+                    'data': {
+                        'file_type': FileType.FILE,
+                        'change_type': ChangeType.MODIFIED,
+                        'modified': None,
+                    },
+                    'min_version': '1.2.4',
+                    'max_version': '1.2.4',
+                },
+                {
+                    'subpath': 'file',
+                    'data': {
+                        'file_type': FileType.FILE,
+                        'change_type': ChangeType.MODIFIED,
+                        'modified': (0, 0),
+                    },
+                    'min_version': '1.2.4',
+                    'max_version': '1.2.4',
+                },
+                {
+                    'subpath': 'chrdev',
+                    'data': {
+                        'file_type': FileType.CHRDEV,
+                        'change_type': ChangeType.ADDED,
+                        'modified': None,
+                    },
+                },
+                {
+                    'subpath': 'fifo',
+                    'data': {
+                        'file_type': FileType.FIFO,
+                        'change_type': ChangeType.ADDED,
+                        'modified': None,
+                    },
+                },
+                {
+                    'subpath': 'hardlink',
+                    'data': {
+                        'file_type': FileType.FILE,
+                        'change_type': ChangeType.ADDED,
+                        'modified': None,
+                    },
+                },
+                {
+                    'subpath': 'symlink',
+                    'data': {
+                        'file_type': FileType.LINK,
+                        'change_type': ChangeType.ADDED,
+                        'modified': None,
+                    },
+                },
+            ],
+        ),
+        (
+            'test-archive2',
+            'test-archive3',
+            [
+                {
+                    'subpath': 'borg_src',
+                    'match_startsWith': True,
+                    'data': {
+                        # TODO: Check/Review why file_type is FILE instead of DIRECTORY
```

Comment:
Did you do that already?

Reply:
I tried the docs and github issues but couldn't find a explanation.

Reply:
So is this a bug we should open an issue for?

Reply:
Sorry I just assumed it's intended behavior. I'll check some more and compare the actual result of borg w/ parse function and get back on this.